firebase.go

```diff
@@ -173,6 +173,16 @@ func (fb *Firebase) doRequest(method string, body []byte) ([]byte, error) {
 	}
 
 	resp, err := fb.client.Do(req)
+	if err == nil && resp.StatusCode == http.StatusTemporaryRedirect {
+		loc, err := resp.Location()
+		if err != nil {
+			return nil, err
+		}
+
+		fb.url = strings.Split(loc.String(), "/.json")[0]
```

Comment:
Yeah that makes sense. I think this is fine. 

The only other way that comes to mind immediately would be to check the index of `/.json` and then doing string slicing, but they both generate the same result - http://play.golang.org/p/OpF3CT5sJh